docs: Add Getting Started guide for Gateway API support (#21908)
cilium-team/cilium: Documentation/network/servicemesh/gateway-api/gateway-api.rst

.. only:: not (epub or latex or html)

    WARNING: You are looking at unreleased Cilium documentation.
    Please use the official rendered version released here:
    https://docs.cilium.io

.. _gs_gateway_api:

*******************
Gateway API Support
*******************

> Cilium supports Gateway API v0.5.1 for below resources, all the upstream conformance tests
> are passed.

Reviewer: Nit: Maybe clarify this, too?

```suggestion
Cilium supports Gateway API v0.5.1 for the resources listed below, and passes `all upstream conformance tests <add_a_link_here>`__.
```

(Or `all related upstream conformance tests` if they're just the tests relating to supported resources?)

Reviewer 2: I think that `all the Core conformance tests, plus the ReferenceGrant extended tests` is the most correct. (This should change in the next release, when ReferenceGrant is included in the core suite).

.. only:: not (epub or latex or html)

    WARNING: You are looking at unreleased Cilium documentation.
    Please use the official rendered version released here:
    https://docs.cilium.io

.. _gs_gateway_api:

*******************
Gateway API Support
*******************

Cilium supports Gateway API v0.5.1 for below features:

- `GatewayClass <https://gateway-api.sigs.k8s.io/api-types/gatewayclass/>`_
- `Gateway <https://gateway-api.sigs.k8s.io/api-types/gateway/>`_
- `HTTPRoute <https://gateway-api.sigs.k8s.io/api-types/httproute/>`_
- `ReferenceGrant <https://gateway-api.sigs.k8s.io/api-types/referencegrant/>`_

.. Note::

    Even though Cilium only supports above resources, the current
    implementation passes all the conformance tests in the Gateway

>     API suite.

Reviewer: Not clear to me what this means (but I'm not familiar with the concepts here, in particular the Gateway API suite).

Author: yeah, this is a little bit confusing, so it's better just remove it.

.. Note::

    Similar to Ingress, Gateway API controller creates a service of
    LoadBalancer type, so your environment will need to support this.

> 

Reviewer: This should probably be in the prerequisites instead
